src/main/java/com/google/sps/servlets/LoadInterviewsServlet.java

```
@@ -0,0 +1,256 @@
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+import javax.servlet.RequestDispatcher;
+import java.util.Optional;
+
+@WebServlet("/load-interviews")
+public class LoadInterviewsServlet extends HttpServlet {
+
+  private AvailabilityDao availabilityDao;
+  private ScheduledInterviewDao scheduledInterviewDao;
+  private Instant instant;
+
+  @Override
+  public void init() {
+    init(new DatastoreAvailabilityDao(), new DatastoreScheduledInterviewDao(), Instant.now());
+  }
+
+  public void init(
+      AvailabilityDao availabilityDao,
+      ScheduledInterviewDao scheduledInterviewDao,
+      Instant instant) {
+    this.availabilityDao = availabilityDao;
+    this.scheduledInterviewDao = scheduledInterviewDao;
+    this.instant = instant;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    int timezoneOffsetMinutes = Integer.parseInt(request.getParameter("timeZoneOffset"));
+    Preconditions.checkArgument(
+        Math.abs(timezoneOffsetMinutes) <= 720,
+        "Offset greater than 720 minutes (12 hours): %s",
+        timezoneOffsetMinutes);
+    ZoneOffset timezoneOffset = convertIntToOffset(timezoneOffsetMinutes);
+    ZonedDateTime day = generateDay(instant, timezoneOffset);
+    ZonedDateTime utcTime = day.withZoneSameInstant(ZoneOffset.UTC);
+    // TODO: Decide how precise this range should be. Must start after or at now, when should end
+    // be?
+    TimeRange range =
+        new TimeRange(utcTime.toInstant(), utcTime.toInstant().plus(6, ChronoUnit.DAYS));
+    List<PossibleInterviewSlot> possibleInterviews =
+        getPossibleInterviewSlots(range, timezoneOffset);
+
+    String date = possibleInterviews.isEmpty() ? "" : possibleInterviews.get(0).date();
```

It sounds like possibleInterviews.isEmpty() should return an error or otherwise be handled, but I don't see it handled at all.

---

On the possibleInterviewTimes jsp page I return "No interviews are available at this time." It's possible for there to be no interviews available, so I wasn't throwing an error but explaining to the user the current situation.

---

Alright I guess.

(The red flag here is assigning what is ostensibly an encoded date to empty string, but no matter I guess. We're time crunched after all)

---

Marking resolved.